tests/foreman/endtoend/test_ui_endtoend.py

```diff
@@ -79,166 +78,195 @@ def test_positive_find_admin_user(self):
-            * Publish content view
-            * Promote content view to both lifecycles
-            * Create a new libvirt compute resource
-            * Create a new subnet
-            * Create a new domain
-            * Create a new hostgroup and associate previous entities to it
-
-        @Feature: Smoke Test
-
-        @Assert: All entities are created and associated.
+    def test_positive_end_to_end(self):
+        """Perform end to end smoke tests using RH and custom repos.
+
+        1. Create a new user with admin permissions
+        2. Using the new user from above
+            1. Create a new organization
+            2. Clone and upload manifest
+            3. Create a new lifecycle environment
+            4. Create a custom product
+            5. Create a custom YUM repository
+            6. Create a custom PUPPET repository
+            7. Enable a RedHat repository
+            8. Synchronize the three repositories
+            9. Create a new content view
+            10. Associate the three repositories to new content view
+            11. Publish content view
+            12. Promote content view to the lifecycle environment
+            13. Create a new activation key
+            14. Add the products to the activation key
+            15. Create a new libvirt compute resource
+            16. Create a new subnet
+            17. Create a new domain
+            18. Create a new hostgroup and associate previous entities to it
+            19. Provision a client
+
+        @Feature: End to End Test
 
+        @Assert: All tests should succeed and Content should be successfully
+        fetched by client.
         """
-        user_name = gen_string('alpha')
-        password = gen_string('alpha')
+        activation_key_name = gen_string('alpha')
+        compute_resource_name = gen_string('alpha')
+        cv_name = gen_string('alpha')
+        domain_name = DOMAIN % gen_string('alpha')
+        hostgroup_name = gen_string('alpha')
+        le_name = gen_string('alpha')
         org_name = gen_string('alpha')
-        env_1_name = gen_string('alpha')
-        env_2_name = gen_string('alpha')
+        password = gen_string('alpha')
         product_name = gen_string('alpha')
-        yum_repository_name = gen_string('alpha')
         puppet_repository_name = gen_string('alpha')
-        cv_name = gen_string('alpha')
-        compute_resource_name = gen_string('alpha')
+        repos = self.sync.create_repos_tree(RHVA_REPO_TREE)
         subnet_name = gen_string('alpha')
-        domain_name = gen_string('alpha')
-        domain = DOMAIN % domain_name
-        hostgroup_name = gen_string('alpha')
+        username = gen_string('alpha')
+        yum_repository_name = gen_string('alpha')
 
-        # Create new user with admin permissions
+        # step 1: Create a new user with admin permissions
         with Session(self.browser) as session:
             make_user(
```

Comment:
You can specify admin rights for user right away if you like:
`make_user(session, username=username, admin=True)`